src/routers/filing.py

```diff
@@ -13,12 +13,14 @@
 
 from starlette.authentication import requires
 
+from .dependencies import verify_user_lei_relation
+
 
 async def set_db(request: Request, session: Annotated[AsyncSession, Depends(get_session)]):
     request.state.db_session = session
 
 
-router = Router(dependencies=[Depends(set_db)])
+router = Router(dependencies=[Depends(set_db), Depends(verify_user_lei_relation)])
 
 
 @router.get("/periods", response_model=List[FilingPeriodDTO])
```

Comment:
Does this work for paths that don't have `/institutions/{lei}/` in them, like `/periods` here? If yes, cool...I think. :)

I was imagining we'd have to [Include an `APIRouter` in another](https://fastapi.tiangolo.com/tutorial/bigger-applications/#include-an-apirouter-in-another) for these institutions endpoints and add the dependency there. Something like...

```python
router = Router(dependencies=[Depends(set_db)])
lei_router = APIRouter(
    prefix="/institutions",
    dependencies=[Depends(verify_user_lei_relation)],
    ...
)
router.include_router(lei_router)
)

# And then refactor the routes to use `lei_router` as there base.

```

...but if we don't and this _just works_, 👍 .

---

It may still be a nice idea to use sub `APIRouter`s considering how nested out URL structure is, but that'd be a separate issue.

Reply:
Ah. I see. `lei` defaults to `None` and the logic is such that it's skipped if `None`.

Reply:
The only route that doesn't have the LEI in it is the /period.  I was going the separate APIRoute direction for those that don't need to verify the lei to user association but instead I made the lei in the dependency optional and if it isn't there, don't check the association.

Reply:
Makes sense. I think that's fine as-is for now. Keeps it simple. We can switch it up if we expand the API to have more non-LEI endpoints or want to use `APIRouter`s for other reasons.